scripts/e2e/e2e.sh

```diff
@@ -135,16 +149,22 @@ start_docker() {
    done
 }
 
+on_exit() {
+  [ "${VM_CREATED}" ] || return 0
```

Comment:
@akutz One question for this part... Will this `return 0` effect the exit code at all? Meaning if something wrong happened early on, the VM was never created (so `VM_CREATED` is not set), but we want the script to exit with a non-zero return code, will this keep that from happening? I don't think it will since this is a `return` not an `exit`, but just wanted to make sure.

Reply:
I've literally no idea. Let's see:

```shell
$ cat <<EOF | bash -s; echo "${?}"
on_exit() { exit 1; }
trap on_exit EXIT
EOF
1
```

```shell
cat <<EOF | bash -s; echo "${?}"
on_exit() { return 1; }
trap on_exit EXIT
EOF
0
```

Looks like the delegate can return a non-zero value and not affect the exit code.